modules/libcom/src/misc/epicsExport.h

```diff
@@ -75,12 +71,24 @@ typedef void (*REGISTRAR)(void);
  *
  * \param typ Object's data type.
  * \param obj Object's name.
- *
- * \note C++ code needs to wrap with @code extern "C" { } @endcode
  */
+#ifdef __cplusplus
+namespace epics {
+    template<typename T>
+    static inline T* ignore_cv_qualifier(volatile T* x) { return const_cast<T*>(x); }
+    template<typename T>
+    static inline T* ignore_cv_qualifier(const T* x) { return const_cast<T*>(x); }
+    template<typename T>
+    static inline T* ignore_cv_qualifier(T* x) { return x; }
+}
+#define epicsExportAddress(typ, obj) \
+    extern "C" { epicsShareExtern typ *EPICS_EXPORT_POBJ(typ,obj); } \
+    epicsShareDef typ *EPICS_EXPORT_POBJ(typ, obj) = reinterpret_cast<typ *>(epics::ignore_cv_qualifier(&obj))
```

Comment:
fyi. when referencing into C++ namespaces from a C macro, best practice is to start from the global namespace be prefixing `::`.  So eg. `epics::ignore_cv_qualifier` would become `::epics::ignore_cv_qualifier``.

Reply:
I gave up on this, going back to C-style cast.